compiler/rustc_lint/src/late.rs

```diff
@@ -30,13 +30,16 @@ use rustc_span::Span;
 use std::any::Any;
 use std::cell::Cell;
 
-/// Extract the [`LintStore`] from [`Session`].
-///
-/// This function exists because [`Session::lint_store`] is type-erased.
-pub fn unerased_lint_store(sess: &Session) -> &LintStore {
-    let store: &Lrc<_> = sess.lint_store.as_ref().unwrap();
-    let store: &dyn Any = &**store;
-    store.downcast_ref().unwrap()
+pub trait SessionExt {
+    fn lint_store(&self) -> &LintStore;
+}
+
+impl SessionExt for Session {
+    fn lint_store(&self) -> &LintStore {
+        let store: &Lrc<_> = self.lint_store.as_ref().unwrap();
+        let store: &dyn Any = &**store;
+        store.downcast_ref().unwrap()
+    }
 }
```

Comment:
Should we keep `unerased_lint_store`, and prefer a comment on `Session::lint_store`?

Reply:
I don't understand how to answer this question. Can you suggest why that's preferable? I've explained this change in the commit message - did you read it?

Reply:
A comment is usually the simplest way to make something more discoverable. Accessing the unerased lint store is exceptional, so I'm not sure this warrants a new trait & impl.

Reply:
How is it exceptional? The lint store is never accessed *without* unerasing because nothing can be done with the erased type.

Reply:
I have a mild preference for the original function:
- A trait seems like overkill, when there is a single method implemented for a single type. The trait suggests generality, i.e. there might be other types that implement the trait.
- The "because `Session::lint_store` is type-erased" comment is useful, and should be preserved even if the trait is kept.

Reply:
Removed the last commit.